test_scripts/Polices/External_UCS_Informing_HMI/019_ATF_N_GetListofPermissions_empty_due_to_invalid_param_entityType_OnAppPermissionChange.lua

```diff
@@ -0,0 +1,142 @@
+-- Requirement summary:
+-- [Policies] [External UCS] SDL informs HMI about <externalConsentStatus> via GetListOfPermissions response
+-- [HMI API] GetListOfPermissions request/response
+-- [HMI API] ExternalConsentStatus struct & EntityStatus enum
+--
+-- Description:
+-- For Genivi applicable ONLY for 'EXTERNAL_PROPRIETARY' Polcies
+-- Check that SDL invalidates notification OnAppPermissionConsent due to invalid value type of parameter entityType
+--
+-- 1. Used preconditions
+-- SDL is built with External_Proprietary flag
+-- SDL and HMI are running
+-- Application is registered and activated
+-- PTU file is updated and application is assigned to functional groups: Base-4, user-consent groups: Location-1 and Notifications
+-- PTU has passed successfully
+-- HMI sends <externalConsentStatus> to SDl via OnAppPermissionConsent (parameter entityType has invalid value, rest of params present and within bounds, EntityStatus = 'ON')
+-- SDL doesn't receive updated Permission items and consent status
+--
+-- 2. Performed steps
+-- HMI sends to SDL GetListOfPermissions (appID)
+--
+-- Expected result:
+-- SDL sends to HMI empty array
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+-- ToDo (vvvakulenko): remove after issue "ATF does not stop HB timers by closing session and connection" is resolved
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local testCasesForPolicyTable = require('user_modules/shared_testcases/testCasesForPolicyTable')
+local commonTestCases = require('user_modules/shared_testcases/commonTestCases')
+
+--[[Local Variables]]
+local params_invalid_data =
+{
+  {param_value = "invalidValue", comment = "String"}
+  {param_value = 1.32 comment = "Float"},
+  {param_value = {}, comment = "Empty table" },
+  {param_value = { entityType = 1, entityID = 1 }, comment = "Non-empty table"},
+  {param_value = -1, comment = "OutOfLowerBound"},
+  {param_value = 129, comment = "OutOfUpperBound" },
+  {param_value = "", comment = "Empty" },
+  {param_value = nil, desc = "Null" }
+}
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFiles()
+commonSteps:DeletePolicyTable()
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_trigger_getting_device_consent()
+  testCasesForPolicyTable:trigger_getting_device_consent(self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+for i = 1, #params_invalid_data do
```

Comment:
@vvvakulenko, because you have loop and for each loop is expected PTU to be successful, after first PTU(UP_TO_DATE) you should trigger again PTU. You can use as example:

  if(i > 1) then
--[[ Preconditions ]]
    commonFunctions:newTestCasesGroup("Preconditions")

    Test["TestStep_trigger_user_request_update_from_HMI_"..params_invalid_data[i].comment] = function(self)    
      testCasesForPolicyTable:trigger_user_request_update_from_HMI(self)
    end
  end